### DISCUSSION TITLE
Get pane information

### DISCUSSION THREAD

**Discussion by noahbald** [zellij-org/zellij#3738] [Ideas]
Hello, it would be useful to me to get information about panes so that I can more precisely do stuff in zellij.

The first thing that would be useful is a way to get information of the panes in the active tab
- Pane name (or dynamic name)
- pid
- program
- Whether it's the current pane
```sh
zellij list-panes
```
Which would print something like `pane-name [pid] program (current?)`
```
zsh [18531] /bin/zsh (current)
nvim [18632] /opt/homebrew/bin/nvim
"Pane #1" [18532] /bin/zsh
```

This would be useful in finding a specific pane and doing something with it. Eg, from a program, find the pane with a matching pid, then focusing on that pane to write to it.

Then it would be useful to get some deeper information on a specific pane
```sh
zellij action query-pane # get info on the current pane
zellij action query-pane 18531 # get info on the pane running the specific process
```
Which would print something in a standard format (e.g toml)
```yaml
name = "zsh"
pid = 18531
program = "/bin/zsh"
y = 3
x = 1
width = 320
height = 100
```

The positioning information would be useful so that we could create a floating pane positioned in relation to the page. Eg, create a pane that neatly covers the left half of the current pane.

**Comment by imsnif** [comment]
Hey, all of this information is available in the plugin API. The Zellij philosophy is to use plugins rather than the CLI for various reasons (mostly since the CLI is not client-aware, and with Zellij being a true multiplayer application, giving this information through the CLI will never be 100% accurate).

EDIT: you can get *some* of this in `zellij action list-clients` though, if it helps you.

**Comment by noahbald** [comment]
Thanks imsnif, this is helpful but I’m not writing a plugin.

In my particular case I want to control panes I’ve created from a neovim instance.
Not sure if I wanna dive in to plugins for this, but good to know!
